### DISCUSSION TITLE
Miryoku zmk extra keys plus quote_with_column swap

### DISCUSSION THREAD

**Discussion by EgorDuplensky** [manna-harbour/miryoku#330] [Q&A] [answered]
Hi,
Thank you for the layout!
I am trying to create a custom_config.h to achieve few things and wanted to understand if I do this correctly.
1) Swap `"` with `;` for base layer, which is necessary to keep positions of Cyrillic letters.
2) Use extra keys of 3x42 keyboard, to have all the Cyrillic letters available on the base layer.

I am using `dao` 3x42 as a baseline, which is basically `corne` 3x42
There is my diff:
```diff
 // Copyright 2021 Manna Harbour
 // https://github.com/manna-harbour/miryoku
 
+#define XXX &none
+
+#define MIRYOKU_LAYER_BASE \
+&kp Q,             &kp W,             &kp E,             &kp R,             &kp T,             &kp Y,             &kp U,             &kp I,             &kp O,             &kp P,             \
+U_MT(LGUI, A),     U_MT(LALT, S),     U_MT(LCTRL, D),    U_MT(LSHFT, F),    &kp G,             &kp H,             U_MT(LSHFT, J),    U_MT(LCTRL, K),    U_MT(LALT, L),     U_MT(LGUI, SEMI),   \
+U_LT(U_BUTTON, Z), U_MT(RALT, X),     &kp C,             &kp V,             &kp B,             &kp N,             &kp M,             &kp COMMA,         U_MT(RALT, DOT),   U_LT(U_BUTTON, SLASH),\
+U_NP,              U_NP,              U_LT(U_MEDIA, ESC),U_LT(U_NAV, SPACE),U_LT(U_MOUSE, TAB),U_LT(U_SYM, RET),  U_LT(U_NUM, BSPC), U_LT(U_FUN, DEL),  U_NP,              U_NP
+
+#define MIRYOKU_LAYOUTMAPPING_CORNE( \
+     K00, K01, K02, K03, K04,      K05, K06, K07, K08, K09, \
+     K10, K11, K12, K13, K14,      K15, K16, K17, K18, K19, \
+     K20, K21, K22, K23, K24,      K25, K26, K27, K28, K29, \
+     N30, N31, K32, K33, K34,      K35, K36, K37, N38, N39 \
+) \
+&kp RBRC  K00  K01  K02  K03  K04       K05  K06  K07  K08  K09  &kp LBRC \
+     XXX  K10  K11  K12  K13  K14       K15  K16  K17  K18  K19  &kp SQT \
+     XXX  K20  K21  K22  K23  K24       K25  K26  K27  K28  K29  XXX \
+                    K32  K33  K34       K35  K36  K37
+
+#define MIRYOKU_KLUDGE_TOPROWCOMBOS_LEFTPINKIE 2 3
+#define MIRYOKU_KLUDGE_TOPROWCOMBOS_LEFTINNERINDEX 3 4
+#define MIRYOKU_KLUDGE_TOPROWCOMBOS_RIGHTINNERINDEX 7 8
+#define MIRYOKU_KLUDGE_TOPROWCOMBOS_RIGHTPINKIE 8 9
+
+#define MIRYOKU_KLUDGE_BOTTOMROWCOMBOS_LEFTPINKIE 26 27
+#define MIRYOKU_KLUDGE_BOTTOMROWCOMBOS_LEFTINNERINDEX 27 28
+#define MIRYOKU_KLUDGE_BOTTOMROWCOMBOS_RIGHTINNERINDEX 31 32
+#define MIRYOKU_KLUDGE_BOTTOMROWCOMBOS_RIGHTPINKIE 32 33
+
+#define MIRYOKU_KLUDGE_THUMBCOMBOS_LEFT 37 38
+#define MIRYOKU_KLUDGE_THUMBCOMBOS_RIGHT 39 40
+
```

I am trying to understand how to use this custom_config.h together with github action workflow parameters. I.e. I want to also use QWERTY for alphas and vi for nav. Will this all be merged in scope of a workflow build, or I have to specify i.e. `#define MIRYOKU_ALPHAS_QWERTY` and `#define MIRYOKU_NAV_VI` in scope of custom_config.h as well and keep the workflow parameters as a default?

**Comment by manna-harbour** [comment, marked as answer]
You can set options in `custom_config.h` or in the workflow, but the workflow takes precedence. I'd suggest using `custom_config.h` for your personal defaults, and the workflow options to override for individual builds.

In the implementation, the workflow appends the corresponding `#define` for alternative layout options to  `custom_config.h`, and leaving as `default` will not append anything for that option. Likewise for the `custom_config` option, which is appended as-is. For local builds you can only use `custom_config.h`.

If you substitute a custom layer it will take precedence over any alternative layout option for that layer. So in this case, using the `alphas` option will have no effect as you are substituting a custom Base, but using a `nav` option will work as you are not substituting a custom Nav.
